graphql-dgs/src/main/java/com/netflix/graphql/dgs/DgsDirective.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs;
+
+import org.springframework.stereotype.Component;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Mark a class as a custom Directive implementation that gets registered to the framework.
+ *
+ * @author Hantsy Bai <hantsy@gmail.com>
+ */
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Component
+public @interface DgsDirective {
+    String name() default "";
```

Comment:
This should be applied in the schema file.

Reply:
You could argue that the directive name is also applied in the schema file 😜 
I still could imagine it could be a nice addition. see my second point.

Reply:
The name is all that's used to register the directive in `graphql-java`, so I think the list of targets isn't actually useful on the annotation itself.